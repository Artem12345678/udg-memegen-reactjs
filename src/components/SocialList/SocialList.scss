.social {
  display: flex;
  list-style-type: none;

  &__item {
    position: relative;
    z-index: 1;
    display: inline-block;
    margin-right: 10px;

    &:last-child {
      margin-right: 0;
    }
  }
}
