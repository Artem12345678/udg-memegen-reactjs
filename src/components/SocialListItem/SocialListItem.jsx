import React from "react";
import PropTypes from "prop-types";

import instagram from "../../assets/instagram.svg";
import github from "../../assets/github.svg";

import "./SocialListItem.scss";

const SocialListItem = ({ url, icon }) => {
  const image = icon === "instagram" ? instagram : github;

  return (
    <a
      className="social__link"
      href={url}
      target="_blank"
      rel="noopener noreferrer"
    >
      <img src={image} alt="socialIcon" />
    </a>
  );
};

SocialListItem.propTypes = {
  url: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired
};

export default SocialListItem;
