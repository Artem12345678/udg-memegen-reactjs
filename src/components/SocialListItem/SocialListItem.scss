.social__link {
  display: flex;
  width: 28px;
  height: 28px;
  transition: border-color 0.25s ease-in-out;
  text-align: center;
  border: 1px solid transparent;
  border-radius: 50%;
  align-items: center;
  justify-content: center;

  &:focus,
  &:active,
  &:hover {
    border-color: #fff;
  }
}
