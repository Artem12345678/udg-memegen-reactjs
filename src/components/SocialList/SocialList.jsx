import React, { Component } from "react";
import SocialListItem from "../SocialListItem/SocialListItem";

import "./SocialList.scss";

class SocialList extends Component {
  state = {
    links: [
      {
        url: "https://www.github.com/Artem12345678",
        icon: "github"
      },
      {
        url: "https://www.instagram.com/ystimenkoartem",
        icon: "instagram"
      }
    ]
  };

  render() {
    const { links } = this.state;

    return (
      <ul className="social">
        {links.map(item => (
          <li key={item.icon} className="social__item">
            <SocialListItem url={item.url} icon={item.icon} />
          </li>
        ))}
      </ul>
    );
  }
}

export default SocialList;
