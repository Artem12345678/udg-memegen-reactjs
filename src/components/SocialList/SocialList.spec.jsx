import React from "react";
import { shallow } from "enzyme";

import SocialList from "./SocialList";
import SocialListItem from "../SocialListItem/SocialListItem";

describe("SocialList.jsx", () => {
  let wrapper;

  const links = [
    {
      url: "https://www.github.com/Artem12345678",
      icon: "github"
    },
    {
      url: "https://www.instagram.com/ystimenkoartem",
      icon: "instagram"
    }
  ];

  beforeEach(() => {
    wrapper = shallow(<SocialList />);
    wrapper.setState({ links });
  });

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("renders <SocialListItem/> components", () => {
    expect(wrapper.find(SocialListItem).exists()).toBe(true);
  });

  it("renders the right amount of <SocialListItem/> components", () => {
    expect(wrapper.find(SocialListItem).length).toBe(links.length);
  });
});
