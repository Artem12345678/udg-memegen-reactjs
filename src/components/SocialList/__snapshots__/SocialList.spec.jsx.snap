// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialList.jsx matches snapshot 1`] = `
<ul
  className="social"
>
  <li
    className="social__item"
    key="github"
  >
    <SocialListItem
      icon="github"
      url="https://www.github.com/Artem12345678"
    />
  </li>
  <li
    className="social__item"
    key="instagram"
  >
    <SocialListItem
      icon="instagram"
      url="https://www.instagram.com/ystimenkoartem"
    />
  </li>
</ul>
`;
